CONTRIBUTING.md

```diff
@@ -46,14 +46,38 @@ $ bundle exec rake console
 
 Tests are very important part of google-cloud-ruby. All contributions should include tests that ensure the contributed code behaves as expected.
 
+To run the unit tests, documentation tests, and code style checks together:
+
+``` sh
+$ rake ci
+```
+
 ### Unit Tests
 
-To run the unit tests, simply run:
+
+The project uses the [minitest](https://github.com/seattlerb/minitest) library, including [specs](https://github.com/seattlerb/minitest#specs), [mocks](https://github.com/seattlerb/minitest#mocks) and [minitest-autotest](https://github.com/seattlerb/minitest-autotest).
+
+To run the unit tests:
 
 ``` sh
 $ rake test
 ```
 
+### Documentation Tests
+
+The project tests the code examples in the each gem's [YARD]()-based documentation.
+
+The example testing functions in a way that is very similar to unit testing, and in fact the library providing it, [yard-doctest](https://github.com/p0deje/yard-doctest), is based on the project's unit test library, [minitest](https://github.com/seattlerb/minitest).
+
+To run the documentation tests:
+
+``` sh
+$ rake jsondoc
```

Comment:
@hxiong388: It does in practice, but the dependency is not in rake, I think because `jsondoc` is pretty slow, and it's common to run `doctest` frequently while debugging the setup.